fastparquet/test/test_writer.py

```diff
@@ -272,6 +283,23 @@ def test_groups_roundtrip(tempdir):
         assert row.b in list(df[(df.a==row.a)&(df.c==row.c)].b)
 
 
+def test_empty_groupby(tempdir):
+    df = pd.DataFrame({'a': np.random.choice(['a', 'b', None], size=1000),
+                       'b': np.random.randint(0, 64000, size=1000),
+                       'c': np.random.choice([True, False], size=1000)})
+    df.loc[499:, 'c'] = True  # no False in second half
+    writer.write(tempdir, df, partition_on=['a', 'c'], file_scheme='hive',
+                 partitions=[0, 500])
+    r = ParquetFile(tempdir)
+    assert r.count == sum(~df.a.isnull())
+    assert len(r.row_groups) == 6
+    out = r.to_pandas()
```

Comment:
The number of files produced is potentially up to 
( nunique(col_x) ) * ( nunique(col_y) ) ... * num row_splits
although some of those pieces may be empty and not get created;
or equal to
sum( len(gb.indices)  for groupby of each row_group ) 

We could do that sum and error over some user-settable limit. For dask-dataframes, you would not want to calculate nuninque across the whole data-set, only do groupby on the sub pandas datafranes.